rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/EvaluatorUtil.java

```diff
@@ -111,6 +111,15 @@ public static <T> InterruptibleFuture<T> runEvaluator(String task, CompletableFu
                 if (!isParametric) {
                     reportInternalError(e, task, client);
                 }
+
+                // CallFailed is a normal response. It means some cases aren't dealt with by the provider
+                // functions. This can be by design, or the function is not working yet completely.
+                // Here we provide a "default" overload for that function that will return the defaultResult.
+                if (isCallFailedException(e.getException())) {
```

Comment:
I don't like this, since it will hide errors for a longer time. End-users get no feedback they are triggering a bug, and are just not getting the behavior they expected, sometimes this is noticible, for example not jumping to a definition. But sometimes it's about a type-check that only reports errors in case something is wrong. An end-user would never notice it was breaking, they were just not getting any type-check information reported.

Reply:
Your original problem, however, is not solved by either solution. However we add a default, in Jeva code add it is now, in util:: LanguageServer as proposed above, or by the language engineer as a default overload, the error in the pattern match will always be silently ignored.

So which solution is the least bad?

Reply:
@DavyLandman and I talked about this. The CallFailed catch is too wide, it would catch all Rascal functions and on all levels of dispatch and recursion. That's not good. So I'm reverting this change and will explain in documentation how users of the Focus functionality must provide defaults with empty sets or lists.